---
output: word_document
---
```{r, echo = FALSE}
  library(knitr)
  opts_chunk$set(echo = FALSE, message = FALSE, warning = FALSE)
```
```{r}
library(magrittr)
library(snpclust)

# fetch hgdp files
paths = paste0('hgdp.', c('zip', 'txt', 'gds')) %>% file.path(getwd(), .)
snpclust:::fetch_hgdp(paths)

pca_by_region = function(region) {
  region_filename = region %>% gsub(' ', '_', .)
  rds_path = paste0('snpclust_', region_filename, '.rds')

  # load or compute pca
  if (!file.exists(rds_path)) {

    # subset to specific region
    subset_paths = paths
    subset_paths[1:2] %<>% paste0('_', region_filename)
    file.copy(paths[1:2], subset_paths[1:2])

    capture.output({
        snpclust:::reduce_hgdp(subset_paths, region_selection = region)
      })

    # save as gds file
    save_hgdp_as_gds(subset_paths)

    # get qc and pca
    snpclust_object <- snpclust(gds = subset_paths[3], only_pca = TRUE,
      n_cores = parallel::detectCores() / 2)
    save(snpclust_object, file = rds_path)

    # remove temporary files
    file.remove(subset_paths)
  }
  get(load(rds_path))
}

# Get variables and plot for Europe subset

snpclust_object = pca_by_region('Europe')
# take last line of qc df and samples-snps columns
dims = snpclust_object$qc %>% `[`(nrow(.), c('Samples', 'SNPs')) %>%
  format(big.mark = ',')

snpclust_object$pca$population %<>%
  gsub('French_Basque', 'Basque', .) %>%
  gsub('North_Italian', 'Italian', .) %>% factor

ggeu <- ggplot_pca(pca = snpclust_object$pca, groups = 'population') +
  ggplot2::scale_x_reverse() + ggplot2::scale_y_reverse() +
  ggplot2::labs(color = 'Population')

# Get variables and plot for North Africa & Middle East subset

snpclust_object_africa = pca_by_region('Middle Est|North Africa')
# take last line of qc df and samples-snps columns
dims_africa = snpclust_object_africa$qc %>%
  `[`(nrow(.), c('Samples', 'SNPs')) %>% format(big.mark = ',')

ggafr <- ggplot_pca(pca = snpclust_object_africa$pca, groups = 'population') +
  ggplot2::labs(color = 'Population')
```

# Authors

Thomas Charlon^1,2^,
Alessandro Di Cara^1^,
Sviatoslav Voloshynovskiy^2^,
Jérôme Wojcik^1^

1: Quartz Bio, Geneva, Geneva 1202, Switzerland

2: Stochastic Information Processing, University of Geneva, Carouge, Geneva 1227, Switzerland

# Title

Replication of the principal component analyses of the human genome diversity panel

# Abstract

## Background

In 2008, several principal component analyses (PCAs) applied on 660,918 single-nucleotide polymorphisms (SNPs) from 938 individuals from 51 worldwide populations of the Human Genome Diversity Panel were published by Li *et al.*. PCAs were applied on subsets of individuals sharing a common geographic origin and showed that in several geographic regions, genome-wide variations of SNPs grouped individuals by populations in the two first principal components.

In this study we replicated the PCAs applied on two geographic subsets, first on individuals from Europe and second on individuals from the Middle East & North Africa.

## Methods

Quality control, feature selection, and PCA were applied on each geographic subset. The results were displayed on the two first principal components and compared to the original figures.

## Results

The replicated figures were found to match closely to the original figures.

## Conclusions

Therefore, the main results were replicated and can be independently reproduced by using publicly available data, source code, and computing environment.

# Keywords

Bioinformatics

Evolutionary/Comparative Genetics

Genomics

# Introduction

Quartz Bio and the Stochastic Information Processing group are involved in the PRECISESADS project (http://www.precisesads.eu/), which aims at reclassifying Systemic Autoimmune Diseases (SADs), a group of chronic inflammatory conditions characterized by the presence of unspecific autoantibodies in the serum and resulting in serious clinical consequences, based on genetic and molecular biomarkers rather than clinical criteria.

In order to use genetic similarities to deliver personalized treatments to patients affected by SADs as well as other diseases, it is important to first understand the genetic structures in healthy populations.

In 2008, Li *et al.*^[1](#ref1)^ showed that although specific world regions have different genetic origins, all revealed population structures in principal component analyses (PCAs). Similar population structures were also observed in studies using other genome-wide variations datasets ^[2](#ref2),[3](#ref3)^.

Li *et al.* applied PCAs on subsets of individuals from two geographic regions, Europe and the Middle East & North Africa and displayed the results on the two first principal components in their article as Figures 2A and B, respectively (with the latter labeled only Middle East).

In an attempt to replicate these two figures, we performed quality control, minor allele frequency filtering, tag SNP selection^[4](#ref4)^, and PCAs on both regional subsets of the SNP microarray data. The PCAs were then displayed on the first two principal components.

# Methods

## Genotype data

The dataset consisted of two files: a zip file including the genotype data of 660,918 SNPs from 1,043 individuals with the annotations of the SNPs, and a text file composed of the annotations of 953 individuals (see Data and software availability).

The annotations of individuals were used to create two subsets of the data. The first contained 157 individuals from Europe and the second contained 163 individuals from the Middle East & North Africa.

## Analysis sets

For each geographic region subset of the data, we verified that no individuals had missing value rates above 3% and excluded SNPs with missing value rates above 1%. An additive genetic model^[4](#ref4)^ was then used to encode each A/B SNP (A/A = 0, A/B = 1, B/B = 2). SNPs with minor allele frequency below 5% were excluded to remove rare variants, which are more prone to genotyping errors. In addition, in order to decrease the required computation time and memory usage, redundant SNPs were removed by applying TagSNP^[5](#ref5)^ (r2 > 0.8, window of 500,000 base pairs). The missing values were imputed by random sampling of each SNP. Then each SNP was centered and scaled to unit variance. All steps were performed using the SNPClust R package v1.0.0^[2](#ref2)^.

For the Europe subset, a total of `r dims$SNPs` SNPs from `r dims$Samples` individuals were selected for analysis. This defines our Europe analysis set.

For the Middle East & North Africa subset, a total of `r dims_africa$SNPs` SNPs from `r dims_africa$Samples` samples were selected for analysis. This defines our Middle East & North Africa analysis set.

For comparison, the supporting online material of Li *et al.* reported that individuals with missing value rates above 2.5% and SNPs with missing value rates above 5% were excluded. Table S1 of Li *et al.* reports that 156 individuals from Europe and 160 from the Middle East & North Africa were used and the supporting online material reports that 642,690 SNPs were used.

## Principal component analyses

PCAs were applied on the two analysis sets and displayed using the SNPClust R package v1.0.0^[2](#ref2)^. Principal component analysis (PCA) is a dimensionality reduction method, which projects SNPs by linear combination to maximize the variance on successive axes, *i.e.* principal components, while constraining the axes to be orthogonal.

The supporting online material of Li *et al.* reports that they first computed the Identity-by-State (IBS) matrix among the 938 individuals by using PLINK^[6](#ref6)^ (version not specified) and then performed PCAs on the IBS matrix for each region separately. In this study PCAs were applied directly on the analysis sets and not on IBS matrices.

# Results

## PCA of the Europe analysis set

The PCA of the Europe analysis set was displayed on the two first principal components ([Figure 1](#fig1)). Individuals were grouped by population and the replicated figure matched closely with Li *et al.*'s Figure 2A.

The explained variance was almost identical, as the replication stated 2.1% in PC1 and 1.6% in PC2 while  Li *et al.*'s Figure 2A stated 2.4% and 1.6% respectively.

```{r}
ggplot2::ggsave('figure1.eps', ggeu)
```

## PCA of the Middle East & North Africa analysis set

The PCA of the Middle East & North Africa analysis set was displayed on the two first principal components ([Figure 2](#fig2)). Individuals were grouped by populations and the replicated figure matched closely with Li *et al.*'s figure 2B.

Two differences from Li *et al.*'s analysis were noted, first the Bedouin and Druze populations exhibited a larger spread on PC1 in the original figure. Second, one Bedouin individual was located with Mozabite individuals, which did not appear in Li *et al.*'s Figure 2B. 

The explained variance was slightly smaller, as the replication stated 3.1% in PC1 and 2.2% in PC2, while Li *et al.*'s Figure 2B stated 5.0% and 2.6% respectively.

```{r}
ggplot2::ggsave('figure2.eps', ggafr)
```

# Discussion

The replicated figures matched closely to the original figures, although two differences appeared when examining the Middle East & North Africa subset: the smaller spread of two populations and the presence of an outlier.

Therefore, the main results were replicated and can be independently reproduced by using publicly available data, source code, and computing environment.

We successfully confirmed that although the two geographic regions studied had different genetic origins, both exhibited populations structures in PCAs.

Understanding the genetic structure of healthy populations will enable us to use genetic similarities to deliver personalized treatments to patients affected by SADs. Using this replication, the PRECISESADS project will be able to compare clusters of patients affected by SADs to clusters of healthy individuals, independently from their ancestry-driven genetic structure^[2](#ref2)^.

# Data and software availability

All data sets are available from HGDP-CEPH. As stated in Li *et al.*^[1](#ref1)^, the data sets are freely available online. Although the links that were provided are now outdated, the two data files are still available at http://www.hagsc.org/hgdp/data/hgdp.zip and ftp://ftp.cephb.fr/hgdp_v3/hgdp-ceph-unrelated.out.

The PCAs were computed and displayed using the previously published R package SNPClust v1.0.0^[2](#ref2)^.

Computing environment in a Docker container is available from: https://hub.docker.com/r/thomaschln/reproducible-hgdp.

Source code of this article and Dockerfile: https://github.com/ThomasChln/reproducible-hgdp.

Archived source code as at time of publication: https://doi.org/10.5281/zenodo.345122

License: GNU General Public License version 3.0

# Ethical statement

The data were previously published^[1](#ref1)^ and approved by ethics committees. No samples were used and records were de-identified.

# Author contributions

Conceptualization: JW SV;

Formal analysis: TC;

Funding acquisition: JW;

Investigation: JW ADC;

Methodology: TC JW;

Project administration: JW;

Software: TC;

Supervision: JW SV;

Validation: TC JW ADC;

Visualization: TC;

Writing - original draft: TC;

Writing - review & editing: JW ADC SV.

# Competing interests

Thomas Charlon, Alessandro Di Cara, and Jérôme Wojcik are employees of Quartz Bio S.A., Switzerland. The authors declare no competing interests related to this commercial affiliation. This does not alter the authors’ adherence to F1000Research policies on sharing data and materials.

# Grant information

Quartz Bio S.A. provided support in the form of salaries for Thomas Charlon, Alessandro Di Cara, and Jérôme Wojcik, but did not have any additional role in the study design, data collection and analysis, decision to publish, or preparation of the manuscript. This work has received support from the EU/EFPIA/ Innovative Medicines Initiative Joint Undertaking PRECISESADS (grant no. 115565).

# Acknowledgments

We thank K. Forner for contributions on the software.

# References

###### {#ref1}
**1.** LI, Jun Z., et al. Worldwide human relationships inferred from genome-wide patterns of variation. science, 2008, 319. Jg., Nr. 5866, S. 1100-1104.

###### {#ref2}
**2.** CHARLON, Thomas, et al. Single Nucleotide Polymorphism Clustering in Systemic Autoimmune Diseases. PloS one, 2016, 11. Jg., Nr. 8, S. e0160270.

###### {#ref3}
**3.** NOVEMBRE, John, et al. Genes mirror geography within Europe. Nature, 2008, 456. Jg., Nr. 7218, S. 98-101.

###### {#ref4}
**4.** CLARKE, Geraldine M, et al. Basic statistical analysis in genetic case-control studies. Nature Protocols, 2011, Vol. 6, no. 2, p. 121-133. DOI 10.1038/nprot.2010.182.

###### {#ref5}
**5.** STRAM, Daniel O. Tag SNP selection for association studies. Genetic epidemiology, 2004, 27. Jg., Nr. 4, S. 365-374.

###### {#ref6}
**6.** PURCELL, Shaun, et al. PLINK: a tool set for whole-genome association and population-based linkage analyses. The American Journal of Human Genetics, 2007, 81. Jg., Nr. 3, S. 559-575.

# Figures legends

###### {#fig1}

**Figure 1. Two first principal components of the Europe analysis set.**

```{r}
#ggeu
```

Visualization of the principal component analysis on `r dims$SNPs` SNPs from `r dims$Samples` individuals from Europe. Individuals from North and South were differentiated in the first principal component and located in the lower and upper sides respectively. Individuals from East and West were differentiated in the second and located in the right and left sides respectively.

###### {#fig2}

**Figure 2. Two first principal components of the Middle East & North Africa analysis set.**

```{r}
#ggafr
```

Visualization of the principal component analysis on `r dims_africa$SNPs` SNPs from `r dims_africa$Samples` individuals from the Middle East & North Africa. Individuals from East and West were differentiated in the first principal component and located in the right and left sides respectively. Individuals from North and South were differentiated in the second and located in the lower and upper sides respectively.
